server/src/test/java/org/elasticsearch/index/mapper/DocumentParserTests.java

```diff
@@ -1319,74 +1316,145 @@ public void testDynamicDateDetectionEnabledWithNoSpecialCharacters() throws IOEx
         assertThat(dateMapper, instanceOf(DateFieldMapper.class));
     }
 
-    public void testDynamicFieldsStartingAndEndingWithDot() throws Exception {
-        BytesReference bytes = BytesReference.bytes(XContentFactory.jsonBuilder().startObject().startArray("top.")
-                .startObject().startArray("foo.")
-                .startObject()
-                .field("thing", "bah")
-                .endObject().endArray()
-                .endObject().endArray()
-                .endObject());
+    public void testFieldName() throws Exception {
+        checkFieldNameAndValidation("root", "branch", "leaf");
```

Comment:
THis testsuite had a lot of copy paste code (horrible) and i wanted to repeat the same tests but with aliases. To avoid copying basically the entire class i extracted the common stuff so all these tests simply pass parameters (the different stuff).